internal/controller/manager/controllermanager.go

```diff
@@ -502,6 +424,14 @@ func (c *ControllerManager) runControllerGroup(namespace string) error {
 			"refresh interval", namespace)
 	}
 
+	// before starting, first successfully check the versions of all pgcluster's in the namespace
+	if err := operatorupgrade.CheckVersion(c.controllers[namespace].pgoRESTClient,
```

Comment:
`reconcileRBAC` is called in two places:

- When a new controller groups is added
- On each namespace refresh interval

This means when we first detect a new namespace we attempt to reconcile RBAC for it.  But in the event that we can't yet (because say some other Kube Admin needs to create the RBAC needed to allow it), we'll then just try again on the next refresh interval.

As for the line referenced here with the `CheckVersion` function, this is simply used to check whether or not any `pgclusters` in the namespace are from an older version of the Operator, and therefore need to be upgraded.